Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs

```diff
@@ -419,6 +419,12 @@ private void Start()
             {
                 constraintsManager = gameObject.EnsureComponent<ConstraintManager>();
             }
+
+            if (!AllowFarManipulation && GetComponent<NearInteractionGrabbable>() == null)
```

Comment:
I don't quite think this is the right approach. Mainly because `NearInteractionGrabbable` missing will be a problem any time a near manipulation is desired. Near manipulations can be desired whether or not far manipulations are enabled. In fact, in my experience both near and far manipulations are desired the majority of the time.

The problem is I don't know a better place to put this. Perhaps we downgrade it to a warning and always show the warning in OnEnable? Something like:

> Warning: Near interactions are not enabled for <GameObjectName> because NearInteractionGrabbable wasn't found.  To enable near interactions with ObjectManipulator, add a NearInteractionGrabbable component.

Reply:
Note also that a NearInteractionGrabbable could also be on a descendant collider of this object, though I do not see that often. For example, say you have:

```
manipulation root (has ObjectManipulator)
|- collider1 (has collider with NearInteractionGrabbable)
|- collider2 (has collider with NearInteractionGrabbable)
```

Then, the manipulation root would be near manipulatable. Here's another structure where the colliders are not immediate children:

```
manipulation root (has ObjectManipulator)
|- child A
   |- collider1 (has collider with NearInteractionGrabbable)
|- child B
   |- child C
      |- collider2 (has collider with NearInteractionGrabbable)
```

In all of these cases, near manipulation would work on the objectmanipulator. The key is that all for each descendant collider of the object, the object with that collider also must have a NearInteractionGrabbable component.

Reply:
Note, the cases I am pointing out are a bit esoteric, in most cases I see people adding a nearinteractiongrabbable directly to the objectmanipulator, and that objectmanipulator has a collider on it.

Reply:
Since the component functions properly if a near interaction grabbable is attached to a child object with a collider, instead of this block: 

```
            if (!AllowFarManipulation && GetComponent<NearInteractionGrabbable>() == null)
            {
                Debug.LogError($"AllowFarManipulation is false and a NearInteractionGrabbable component is not attached to {gameObject.name}. " +
                " To enable near interaction support, attach a NearInteractionGrabbable component to this object.");
            }
```

we could check for a NearInteractionGrabbable in the child objects:

```
            var children = GetComponentsInChildren<NearInteractionGrabbable>();

            if (!AllowFarManipulation && children.Length == 0)
            {
                Debug.LogWarning($"Near interactions are not enabled for {gameObject.name}. To enable near interactions, add a " +
                    $"NearInteractionGrabbable component to {gameObject.name} or to a child object of {gameObject.name} that contains a collider.");
            }
```

Reply:
Also great catch @julenka. @CDiaz-MS I like the approach (assuming the check will happen regardless of `AllowFarManipulation`). One nit / suggestion would be to use `{nameof(NearInteractionGrabbable)}` rather than the hardcoded text so that it survives any refactoring.